src/SessionManager/ProgramSettings.cc

```diff
@@ -76,7 +76,8 @@ ProgramSettings::ProgramSettings(int argc, char** argv) {
  3   warning
  4   info
  5   debug
- 6   trace)");
+ 6   trace
+ 7   trace + ICD)");
```

Comment:
each level implies all the previous levels. I suggest you just say "ICD". I also think we should rename "trace" to `performance`. We can use `#defines` to customise the `spdlog::trace` prefix as well

Reply:
the one problem with that is that we're still calling `spdlog::trace` when logging performance. We can add a function alias though:
```
// Adds an alias to trace for consistency between function name and prefix
namespace spdlog {
constexpr auto perflog = [](auto&& ...args) {
    return trace(std::forward<decltype(args)>(args)...);
};
```

Reply:
So should we rename all the `spdlog::trace` as `spdlog::performance` in the original code. And add a new customize `spdlog::icd` as well?

Reply:
I think we should rename the existing `spdlog::trace`, yes. Not sure about adding a new ICD, since we only use it once.

Reply:
This alias isn't reflected in the logged level names, right?

Reply:
Another way is that we can create a new logger and customize its format. I guess.

Reply:
we need to do that too. Something like 
```#define SPDLOG_LEVEL_NAMES {"performance", "debug", "info", "warning", "error", "critical", "off" }```

_before_ including `spdlog.h` would do it